### PR TITLE
[Destruction] Flashover + Soul Fire (and Darkglare fix)

### PR DESCRIPTION
**Pull request by sMteX**
Adds bonus damage from Flashover, together with bonus/wasted Backdraft stacks. Adds Soul Fire damage + its CD reduction via spending shards. Also fixes event prop in Darkglare module from Affli.

Flashover example log: /report/7aKQwdf4M86BcGgx/46-Mythic+Zul+-+Kill+(3:23)/83-Strungol
Soul Fire example log: /report/kGJgV18NAX7Hzwac/10-LFR+Taloc+-+Kill+(4:24)/92-Stanislâs

### CODE REVIEW

File: src/parser/warlock/destruction/modules/core/SpellUsable.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  on_byPlayer_spendresource(event) {
+    if (!this.hasSF) {
+      return;
+    }
+    if (this.isOnCooldown(SPELLS.SOUL_FIRE_TALENT.id)) {
+      // event.resourceChange is in multiples of 10 (2 shards = 20)
+      const shardsSpent = event.resourceChange / 10;
+      this.reduceCooldown(SPELLS.SOUL_FIRE_TALENT.id, shardsSpent * REDUCTION_MS_PER_SHARD);
+    }
+  }
```
Comment (Pewtro) [inline src/parser/warlock/destruction/modules/core/SpellUsable.js]: Just confirming here - shards **only** show in multiples of 10 in logs?

Comment (sMteX) [reply src/parser/warlock/destruction/modules/core/SpellUsable.js]: I can check a few logs to be sure but afaik yes, 20 or 30.

File: src/parser/warlock/destruction/modules/talents/Flashover.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+      // We don't waste the base Backdraft stack, nor the bonus one
+      this._currentStacks += STACKS_PER_CAST;
+      this.bonusStacks += 1;
+    }
+    else {
+      // Conflagrate grants 2 stacks with Flashover, we can attribute 1 to the base Conflagrate and 1 to the Flashover, in this case the Flashover one would get wasted
+      debug && this.log('WASTE');
+      this._currentStacks = MAX_STACKS;
+      this.wastedStacks += 1;
+    }
```
Comment (Pewtro) [inline src/parser/warlock/destruction/modules/talents/Flashover.js]: Is it possible to waste both stacks?

Comment (sMteX) [reply src/parser/warlock/destruction/modules/talents/Flashover.js]: It is, when you Conflagrate on 4 stacks , but I only count the bonus one from the talent (the other wasted stack is "base"). Counting of all wasted stacks is done in Backdraft module.

File: src/parser/warlock/destruction/modules/core/SpellUsable.js (hunk 1 above)
Comment (Pewtro) [reply src/parser/warlock/destruction/modules/core/SpellUsable.js]: I'm fairly sure it is, but once you've checked to be sure you can hit the merge button.

Comment (sMteX) [reply src/parser/warlock/destruction/modules/core/SpellUsable.js]: Looked through 3 logs with Soul Fire (smh, only after 1000th place on LFR Taloc), all spendresource events had resourceChange in multiples of 10.

**Pull request merged**